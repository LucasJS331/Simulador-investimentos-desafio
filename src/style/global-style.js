import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
    *{
        box-sizing: border-box;
        padding: 0;
        margin: 0;
        font-family: 'arial';
    }
    html{
        font-size: 62.5%;
        scroll-behavior: smooth;
    }
    body{
        font-size: 1.6rem;
    }
    input {
      display: block;
      outline: none;
      border: none;
      border-bottom: 1px solid black;
      background-color: transparent;
      padding: 0.5rem;
      width: 100%;
    }

`;
