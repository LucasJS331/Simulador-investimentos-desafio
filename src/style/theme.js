export const theme = {
  colors: {
    primary_color: '#E5E8EB',
    secondary_color: '#FF7D48',
    secondary_color_hover: '#E16736',
  },
  media: {
    default: '(max-width: 830px)',
  },
};
