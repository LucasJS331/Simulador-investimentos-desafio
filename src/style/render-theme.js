import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { FinantialContext } from '../context/finantial-context';
import { theme } from './theme';

export const renderTheme = (children) => {
  return render(
    <FinantialContext>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </FinantialContext>,
  );
};
