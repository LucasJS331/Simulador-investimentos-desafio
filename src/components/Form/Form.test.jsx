import { Form } from '.';
import { renderTheme } from '../../style/render-theme';
const form_mock = [
  {
    valor: 99,
  },
  {
    valor: 69,
  },
];
describe('<Form/>', () => {
  it('should render form', () => {
    const { container } = renderTheme(<Form indicadores={form_mock} />);
    expect(container).toMatchSnapshot();
  });
});
