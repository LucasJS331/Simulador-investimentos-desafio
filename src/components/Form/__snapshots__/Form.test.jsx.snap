// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> should render form 1`] = `
<div>
  <form
    action="#"
    class="sc-pVTFL crJtVe"
  >
    <fieldset>
      <div
        class="sc-dkPtRN cNwulK"
      >
        <h3
          id="rendimento-title"
        >
          Rendimento
        </h3>
        <a
          class="rendimento"
          href="#"
        >
          Bruto
        </a>
        <a
          class="rendimento-bg-br-10"
        >
          Liquido
        </a>
      </div>
      <label
        class="sc-furwcr hlwOoS"
      >
        Aporte inicial
      </label>
      <div
        class="sc-bdvvtL hcYmyj"
      >
        <input
          class=""
          name="aporte"
          value=""
        />
        <label
          class="of"
        >
          
           
        </label>
      </div>
      <label
        class="sc-furwcr hlwOoS"
      >
        Prazo (em meses)
      </label>
      <div
        class="sc-bdvvtL hcYmyj"
      >
        <input
          class=""
          name="prazo"
          value=""
        />
        <label
          class="of"
        >
          
           
        </label>
      </div>
      <label
        class="sc-furwcr hlwOoS"
      >
        IPCA(ao ano)
      </label>
      <div
        class="sc-bdvvtL hcYmyj"
      >
        <input
          class=""
          name="ipca"
          value="69"
        />
        <label
          class="of"
        >
          
           
        </label>
      </div>
    </fieldset>
    <fieldset>
      <div
        class="sc-hKwDye crhrtS"
      >
        <h3
          class="index-title"
        >
          Tipos de Indexação
        </h3>
        <a
          class="btn-selected-bg"
          href="#"
        >
          PRÉ
        </a>
        <a
          class=" btn-selected-br0 "
        >
          PÓS
        </a>
        <a
          class="btn-selected-br10"
        >
          FIXADO
        </a>
      </div>
      <label
        class="sc-furwcr hlwOoS"
      >
        Aporte mensal
      </label>
      <div
        class="sc-bdvvtL hcYmyj"
      >
        <input
          class=""
          name="aporte_mensal"
          value=""
        />
        <label
          class="of"
        >
          
           
        </label>
      </div>
      <label
        class="sc-furwcr hlwOoS"
      >
        Rentabilidade
      </label>
      <div
        class="sc-bdvvtL hcYmyj"
      >
        <input
          class=""
          name="rentabilidade"
          value=""
        />
        <label
          class="of"
        >
          
           
        </label>
      </div>
      <label
        class="sc-furwcr hlwOoS"
      >
        CDI(ao ano)
      </label>
      <div
        class="sc-bdvvtL hcYmyj"
      >
        <input
          class=""
          name="cdi"
          value="99"
        />
        <label
          class="of"
        >
          
           
        </label>
      </div>
    </fieldset>
    <fieldset />
    <button
      class="sc-gsDKAQ BxhBY"
      type="reset"
    >
      Limpar Campos
    </button>
    <a
      class="sc-iCfMLu cVKiPP"
      id="simulador-btn"
    >
      Simular
    </a>
  </form>
</div>
`;
